core-bundle/src/Resources/contao/classes/DataContainer.php

```diff
@@ -409,7 +409,7 @@ protected function row($strPalette=null)
 				}
 			}
 			// Use strlen() here (see #3277)
-			elseif (!\strlen($this->varValue))
+			elseif ('' !== (string) $this->varValue)
```

Comment:
Comment above needs to be adjusted. Or maybe use `?: '' `?

Reply:
I would prefer to remove the comment since its pretty useless,  is meant for contao/core and itself references several other issues. Ok with you @leofeyer ?

Reply:
removed in 9c6f417e2fab70fa4b6b64a2835396125e73d5b9

Reply:
Not sure about this. The comment explicitly says to use `strlen()`…

Reply:
`!\strlen(…)` would be the same same as `'' === (string) …`. This change would reverse the logic because it uses `!==` instead of `===`.

To get rid of the warning we could also use:

```suggestion
			elseif (!\strlen((string) $this->varValue)) 
```

Reply:
I‘m fine with replacing `strlen()` if we are 100% sure that the behavior does not change. 👍 Obviously, we need to fix the `!==` then.

Reply:
I’m 100% sure that `!\strlen((string) …)` is exactly the same same as `'' === (string) …`

Reply:
so why do we need to change anything on the PR if the current is exactly the same as the suggested change? The comment intended to say we must compare a string, not we have to use a specific function …

Reply:
It is not the same? You have reversed the logic in your changes! See @ausi‘s comment above.

Reply:
Aah, right, thanks 😊 
fixed in bcfe39a1265ae7c65b87fa41ab181866056faceb then